src/skmatter/metrics/_prediction_rigidities.py

```diff
@@ -0,0 +1,199 @@
+        ndarray.
+
+    X_test : list of ndarray of shape (n_atoms, n_features)
+        Test dataset where each training set structure is stored as a separate 
+        ndarray.        
+
+    reg : float value
+        Regularizer value that the linear/kernel model has been optimized to.
+
+    Returns
+    -------
+    LPR : list of arrays of shape (n_atoms)
+    rank_diff : integer value of the difference between cov matrix dimension and rank        
+
+    """
+
+    # initialize a StandardFlexibleScaler and fit to train set atom envs
+    scaler = StandardFlexibleScaler()
+    X_atom = np.vstack(X_train)
+    scaler.fit(X_atom)
+
+    # prep to build covariance matrix XX, take average feat vecs per struc
+    X_struc = []
+    for i, cur_X in enumerate(X_train):
+        X_struc.append(np.mean(scaler.transform(cur_X), axis=0))
+    X_struc = np.vstack(X_struc)
+
+    # build XX and obtain Xinv for LPR calculation
+    XX = X_struc.T @ X_struc
+    Xprime = XX + reg * np.eye(XX.shape[0])
+    rank_diff = X_struc.shape[1] - np.linalg.matrix_rank(Xprime)
+    Xinv = np.linalg.inv(Xprime)
+
+    # track test set atom indices for output
+    lens = []
+    for X in X_test:
+        lens.append(len(X))
+    test_idxs = np.cumsum([0] + lens)    
+
+    # prep and compute LPR
+    num_test = len(X_test)
+    X_test = np.vstack(X_test)
+    atom_count = X_test.shape[0]
+    LPR_np = np.zeros(X_test.shape[0])
+
+    for ai in range(atom_count):
+        Xi = scaler.transform(X_test[ai].reshape(1, -1))
+        LPR_np[ai] = 1 / (Xi @ Xinv @ Xi.T)
+
+    # separately store LPR by test struc
+    LPR = []
+    for i in range(num_test):
+        LPR.append(LPR_np[test_idxs[i]:test_idxs[i+1]])
+
+    return LPR, rank_diff
+
+
+def componentwise_prediction_rigidity(
+    X_train,
+    X_test,
+    reg,
+    comp_dims
+):
+    r"""Computes the component-wise prediction rigidity (CPR) and the local CPR 
+    (LCPR) of a linear model trained on a training dataset provided as input, 
+    on the local environments in the test set provided as a separate input.
+    CPR and LCPR are defined as follows:
+
+    .. math::
+        CPR_{A,c} = \frac{1}{X_{A,c} @ (XX + \lambda I)^{-1} @ X_{A,c}^{T}}
+        LCPR_{i,c} = \frac{1}{X_{i,c} @ (XX + \lambda I)^{-1} @ X_{i,c}^{T}}
+
+
+    The function assumes that the feature vectors for the local environments and 
+    structures are built by concatenating the descriptors of different prediction
+    components together. It also assumes, like the case of LPR, that model training
+    is undertaken in a manner where the global prediction targets are averaged over
+    the number of atoms appearing in each training structure, and the average feature
+    vector of each structure is hence used in the regression. Likewise, to guarantee
+    valid comparison in the (L)CPR across different models, feature vectors are scaled
+    by a global factor based on standard deviation across atomic envs.
```

Comment:
I agree that in theory, this can be applied outside of atomistic learning. Yet, I find it hard to explain the LPR and CPR in a purely theoretical sense. On top of that, as this has been written for scikit-_matter_, I think we should be fine just keeping the current doc that focuses its explanations on atomistic learning.